### PR TITLE
Vyper Plugin Patch/Ui Updates

### LINKED ISSUE #4536: Vyper compiler error bugs
1. make an error in a .vy file
2. compile - see error card & error warning badge on the Vyper icon in the icon's panel
3. fix error
4. compile
5. badge says compile successful but the error card is still there - and no compilation details button

### LINKED ISSUE #4537: Vyper checklist
- [x] https://github.com/ethereum/remix-project/issues/4536
- [x] Make an accordion called Advanced configurations (just like solidity) move the radio buttons for remote & local compiler in there.

- [x] Fix Remote compiler Radio button label so it is clickable and make sure both have a cursor pointer (the finger icon)
- [x] Make sure that compiler version & EVM version are in the compilation details
- [x] Remove the Vyper word with the new logo at the top of the plugin
- [x] Add the link to the docs on the text "Specify the compiler version & EVM version in the .vy file" and put a period at the end of the text. Add the link to https://docs.vyperlang.org/en/stable/compiling-a-contract.html#setting-the-target-evm-version on `EVM` and the link (if they update the vyper docs) to the appropriate link for the compiler version on the text `compiler version`. Or these links could go to our Vyper compiler docs.
- [x] Add the Ask GPT button to the compiler error
- [x] Make play button work for compilation work
- [x] ctrl + s for compilation on a .vy file 
- [x]  replace vyper-lang for cloning with this repo: https://github.com/pcaversaccio/snekmate 
- [x] write vyper compiler docs

### PR DESCRIPTION
**Pull request by joeizang**
fixes #4536 #4537

### CODE REVIEW

**Review by ryestew — CHANGES_REQUESTED** [review]

File: apps/vyper/src/app/app.tsx [hunk 1]
```diff
@@ -117,13 +114,30 @@ const App = () => {
...
-          <div className="d-flex flex-row gap-5 mb-3 mt-2">
-            <Form.Check inline data-id="remote-compiler" type="radio" value={state.environment} checked={state.environment === 'remote'} onChange={() => setEnvironment('remote')} label="Remote Compiler" style={{cursor: state.environment === 'remote' ? 'default' : 'pointer'}} className="d-flex mr-4" />
-            <Form.Check inline id="local-compiler" data-id="local-compiler" checked={state.environment === 'local'} type="radio" name="local" value={state.environment} onChange={() => setEnvironment('local')} label="Local Compiler" style={{cursor: state.environment === 'local' ? 'default' : 'pointer'}} />
+
+        <Accordion className="border-0 w-100 mb-3 accordion-background">
+          <div className="border-0">
+            <div className="pl-3 d-flex flex-row justify-content-between">
+              <CustomAccordionToggle eventKey="0">
+                <span className="">Advanced Compiler Settings</span>
+              </CustomAccordionToggle>
```
[the remark this answers is not in the document]
Comment (joeizang) [reply apps/vyper/src/app/app.tsx:123]: done

**Review by ryestew — COMMENTED** [review]
It would make a little more graphic sense if when you click the local compiler, the text box appears just below the radio buttons and not below the the advice text about where to put the pragma stuff.

I"ll have the link for the compiler version today. 

Will the snekmate repo be in this release - or the next one?

Is Ctrl +S working now?  I think it isn't

and the evm version isnt being returned yet from Ape - right?  Nor is the optimize - right?

**Comment by joeizang** [comment]
> It would make a little more graphic sense if when you click the local compiler, the text box appears just below the radio buttons and not below the the advice text about where to put the pragma stuff.
> 
> I"ll have the link for the compiler version today.
> 
> Will the snekmate repo be in this release - or the next one?
> 
> Is Ctrl +S working now? I think it isn't
> 
> and the evm version isnt being returned yet from Ape - right? Nor is the optimize - right?

evm version isn't being returned by Apeworx yet.

Ctrl + S isn't working yet. That will come in the next release.

I will fix the placement of the warning message. 

Thanks @ryestew

**Review by ryestew — CHANGES_REQUESTED** [review]
- [x] For specifying the Vyper version, add this link
https://remix-ide.readthedocs.io/en/latest/vyper.html#specify-vyper-version

- [ ] The tab for compilation details doesn't say the contract's name - the title of the tab is contractName.  This needs to be updated.
- [x] When you switch to another vyper file, the compilation details button should go away - just like it does in Solidity file.
- [ ] I am not getting ctrl +s to work
- [ ] When I click on a compiler error, the cursor should jump to the that line in the editor.  That doesn't seem to be working now.

**Review by ryestew — APPROVED** [review]

**Pull request merged**